src/Extensions/Logging/Internal/EventSource/OmexLogEventSender.cs

```diff
@@ -0,0 +1,182 @@
+			switch (level)
+			{
+				case LogLevel.None:
+					break;
+				case LogLevel.Trace:
+					m_eventSource.LogSpamServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Spam", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Debug:
+					m_eventSource.LogVerboseServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Verbose", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Information:
+					m_eventSource.LogInfoServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Info", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Warning:
+					m_eventSource.LogWarningServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Warning", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Error:
+				case LogLevel.Critical:
+					m_eventSource.LogErrorServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Error", category, tagId, eventId.Name, threadId, message);
+					break;
+				default:
+					throw new ArgumentException("Unknown EventLevel: " + level);
+			}
+		}
+
+
+		public bool IsEnabled(LogLevel level) =>
+		level switch
+		{
+			LogLevel.None => false,
+			_ => m_eventSource.IsEnabled()
+		};
+
+
+		public bool IsReplayableMessage(LogLevel logLevel) =>
+			logLevel switch
+			{
+				LogLevel.Trace => true,
+				LogLevel.Debug => true,
+				_ => false
+			};
+
+
+		public void ReplayLogs(Activity activity)
+		{
+			if (activity is ReplayableActivity replayableActivity)
+			{
+				foreach (LogMessageInformation log in replayableActivity.GetLogEvents())
+				{
+					LogMessage(activity.Id, activity.TraceId, log.Category, LogLevel.Information, log.EventId, log.ThreadId, log.Message);
+				}
+			}
+		}
+
+
+		private readonly OmexLogEventSource m_eventSource;
+		private readonly IServiceContext m_serviceContext;
+		private readonly IMachineInformation m_machineInformation;
+		private static readonly string s_processName;
+
+
+		/// <summary>
+		/// Get the Tag id as a string
+		/// </summary>
+		/// <param name="tagId">tag id</param>
+		/// <returns>the tag as string</returns>
+		/// <remarks>
+		/// In terms of the conversion from integer tag value to equivalent string reprsentation, the following scheme is used:
+		/// 1. If the integer tag &lt;= 0x0000FFFF, treat the tag as special tag called numeric only tag.
+		/// Hence the string representation is direct conversion i.e. tag id 6700 == 6700
+		/// 2. Else, if it's an alphanumeric tag, there are 2 different schemes to pack those. viz. 4 letter and 5 letter representations.
```

Comment:
What's "viz." about?

Reply:
I don't know. This method is a copy from our previous three loggers and I don't really want to change it until we'll make sure that other parts of the system are working